lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2134,37 +2134,51 @@ namespace Js
     Var JavascriptString::ToCaseCore(JavascriptString* pThis, ToCase toCase)
     {
         charcount_t count = pThis->GetLength();
+        charcount_t countToCase = count;
 
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* inStrLim = inStr + count;
```

Comment:
Alright, will do. Possibly next weekend though, and I will conduce some more experiments with an extensive test case, and a perf worse-case scenario ;-)